Source/monster.cpp

```diff
@@ -1095,92 +1095,72 @@ void MonsterHitMonster(int mid, int i, int dam)
 	}
 }
 
-void StartMonsterDeath(int i, int pnum, bool sendmsg)
+void MonsterDeath(int mid, int pnum, Direction md, bool sendmsg)
 {
-	assert(i >= 0 && i < MAXMONSTERS);
-	auto &monster = Monsters[i];
+	assert(mid >= 0 && mid < MAXMONSTERS);
+	auto &monster = Monsters[mid];
 	assert(monster.MType != nullptr);
 
-	if (pnum >= 0)
-		monster.mWhoHit |= 1 << pnum;
-	if (pnum < MAX_PLRS && i >= MAX_PLRS) /// BUGFIX: i >= MAX_PLRS (fixed)
-		AddPlrMonstExper(monster.mLevel, monster.mExp, monster.mWhoHit);
+	if (pnum < MAX_PLRS) {
+		if (pnum >= 0)
+			monster.mWhoHit |= 1 << pnum;
+		if (pnum < MAX_PLRS && mid >= MAX_PLRS)

```

Comment:
The outer condition makes this one unnecessary.
```suggestion
		if (mid >= MAX_PLRS)
```